src/client/dfs/dfs.c

```diff
@@ -1249,7 +1249,12 @@ open_file(dfs_t *dfs, dfs_obj_t *parent, int flags, daos_oclass_id_t cid,
 			/** just try fetching entry to open the file */
 			daos_array_close(file->oh, NULL);
 		} else if (rc) {
-			daos_array_close(file->oh, NULL);
+			int rc2;
+
+			rc2 = daos_array_close(file->oh, NULL);
```

Comment:
i really hate these changes in the code just to support fault injection. i think it's time we figure some other solution rather than polluting the code with these changes.

Reply:
Separately, I thought daos_array_close() was a local operation, does it need to be as complicated as it is?